src/Microsoft.AspNetCore.Mvc.Razor/DependencyInjection/MvcRazorMvcCoreBuilderExtensions.cs

```diff
@@ -74,9 +74,9 @@ private static void AddRazorViewEngineFeatureProviders(IMvcCoreBuilder builder)
                 builder.PartManager.FeatureProviders.Add(new TagHelperFeatureProvider());
             }
 
-            if (!builder.PartManager.FeatureProviders.OfType<ViewsFeatureProvider>().Any())
+            if (!builder.PartManager.FeatureProviders.OfType<CompiledViewsFeatureProvider>().Any())
             {
-                builder.PartManager.FeatureProviders.Add(new ViewsFeatureProvider());
+                builder.PartManager.FeatureProviders.Add(new CompiledViewsFeatureProvider());
```

Comment:
If we're making a new thing, can we call it `CompiledRazorItemFeatureProvider`. It's not views, it's views and pages.

Reply:
Also it's a breaking change not to register `ViewsFeatureProvider`

Reply:
> Also it's a breaking change not to register ViewsFeatureProvider

I assumed users would only need to care about the actual feature rather than the presence of feature providers. Can we add the equivalent of a compat switch to `ViewsFeatureProvider` so it no-ops by default if it does need to be present? I'd like to get rid of the type eventually if we expect startup discovery to discover view assemblies.